htdocs/contact/ajax/contact.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/* Copyright (C) 2006      Andre Cianfarani     <acianfa@free.fr>
+ * Copyright (C) 2005-2012 Regis Houssin        <regis.houssin@inodbox.com>
+ * Copyright (C) 2007-2011 Laurent Destailleur  <eldy@users.sourceforge.net>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ *       \file       htdocs/contact/ajax/contact.php
+ *       \brief      File to return Ajax response on contact list request
+ */
+
+if (! defined('NOTOKENRENEWAL')) define('NOTOKENRENEWAL', 1); // Disables token renewal
+if (! defined('NOREQUIREMENU'))  define('NOREQUIREMENU', '1');
+if (! defined('NOREQUIREHTML'))  define('NOREQUIREHTML', '1');
+if (! defined('NOREQUIREAJAX'))  define('NOREQUIREAJAX', '1');
+if (! defined('NOREQUIRESOC'))   define('NOREQUIRESOC', '1');
+if (! defined('NOCSRFCHECK'))    define('NOCSRFCHECK', '1');
+
+require '../../main.inc.php';
+
+$htmlname=GETPOST('htmlname', 'alpha');
+$outjson=(GETPOST('outjson', 'int') ? GETPOST('outjson', 'int') : 0);
+$action=GETPOST('action', 'alpha');
+$id=GETPOST('id', 'int');
+$socid=GETPOST('socid', 'int');
+$showempty=GETPOST('showempty', 'int');
+$showfunction=GETPOST('showfunction', 'int');
+$showsoc=GETPOST('showsoc', 'int');
+$exclude=GETPOST('exclude', 'array');
+$page=GETPOST('page', 'int');
+$limit=GETPOST('limit', 'int');
+
+
```

Comment:
It seems anybody even without permission can scan and download all the contact database with this new file.
Can you 
- force $socid if user->socid is set (for external users)
- test that user as permission to read thirdparties
- Also if the user has not the permission "Read every thirdparty", a join must be added into the new selectcontactlist to limit the select to contact linked to thirdparty that as the user as sale representative so he can get only list of contact of thirdparties he is allowed to read.